sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -91,14 +91,38 @@ static void *_event_loop(void *arg)
 
         if (res > 0) {
             switch (msg_rcvd.type) {
-                case GCOAP_MSG_TYPE_TIMEOUT:
-                    _expire_request((gcoap_request_memo_t *)msg_rcvd.content.ptr);
-                    break;
-                case GCOAP_MSG_TYPE_INTR:
-                    /* next _listen() timeout will account for open requests */
-                    break;
-                default:
-                    break;
+            case GCOAP_MSG_TYPE_TIMEOUT: {
+                gcoap_request_memo_t *memo = (gcoap_request_memo_t *)msg_rcvd.content.ptr;
+
+                /* no retries remaining */
+                if ((memo->send_limit == GCOAP_SEND_LIMIT_NON)
+                        || (memo->send_limit == 0)) {
+                    _expire_request(memo);
+                }
+                /* reduce retries remaining, double timeout and resend */
+                else {
+                    memo->send_limit--;
+                    unsigned i        = COAP_MAX_RETRANSMIT - memo->send_limit;
+                    uint32_t timeout  = ((uint32_t)COAP_ACK_TIMEOUT << i) * US_PER_SEC;
+                    uint32_t variance = ((uint32_t)COAP_ACK_VARIANCE << i) * US_PER_SEC;
+                    timeout = random_uint32_range(timeout, timeout + variance);
+
+                    ssize_t res2 = sock_udp_send(&_sock, memo->msg.data.pdu_buf,
```

Comment:
maybe `bytes` or `bytes_sent` would be a better name than `res2`? Otherwise, looks good, please squash the fixup.